### PR TITLE
enable PINT to get tdb_tt correction from ephemeris

### PR DESCRIPTION
**Pull request by luojing1211**
This is a pull request to enable PINT read tdt-tt from ephemeris. It is not fully tested yet. If any developer wants to help test this PR, it would be great.

### CODE REVIEW

File: pint/solar_system_ephemerides.py [hunk 1]
```diff
@@ -163,3 +163,46 @@ def objPosVel(obj1, obj2, t, ephem):
         return obj2pv - obj1pv
     else:
         return PosVel(np.zeros((3,len(t)))*u.km, np.zeros((3,len(t)))*u.km/u.second)
+
+def get_tdb_tt_ephem(t, ephem, path=None, link=None):
+    """The is a function to read the TDB_TT correction from the JPL DExxxt.bsp
+       ephemeris file.
```
Comment (paulray) [inline pint/solar_system_ephemerides.py]: Could you add a hyperlink here in the docstring to where the TDB_TT part of the DExxxt ephemeris is defined?

**Comment by paulray** [comment]
This is an excellent addition!

File: pint/solar_system_ephemerides.py (hunk 1 above)
Comment (luojing1211) [inline pint/solar_system_ephemerides.py]: There seems no link for the tt-tdb ephemeris definition. There may be some pdf or txt documents. But the comments in the ephemeris have the definition of segments, which has part of TDB-TT.
If you do:
```
>>> from jplephem.spk import SPK
>>> kernel = SPK.open('de436t.bsp')
>>> print(kernel)
```
It should have the information you need.

**Comment by paulray** [comment]
Does one of the tests compare the values computed from the ephemeris with what astropy gets when using the same ephemeris?  How different are those?  This may not need to be a unit test, but should be done.

**Comment by paulray** [comment]
I'm concerned that this new method may be dropping the topocentric part of the TT to TDB correction.
When we do it with astropy:
```
               grpmjds = time.Time(grp['mjd'], location=grp['mjd'][0].location)
               grptdbs = grpmjds.tdb
```
It knows the location of the observatory, and the topocentric part of the correction is applied (see description [here](https://github.com/nanograv/PINT/wiki/Clock-Corrections-and-Timescales-in-PINT)).
But, the TT_TDB column in the ephemeris is evaluated for the Geocenter.  So, I think that this code will not get the topocentric part correct. Thus, I predict that if you compare these results for DE430 using astropy and the TT_TDB column, you will see a couple of microsecond difference.

**Comment by luojing1211** [comment]
I am doing the test right now. The difference seem in the microsecond level. But I have to double check if I am doing the right thing.

**Comment by luojing1211** [comment]
I have the difference between those two method. I can see the microsecond difference. 
screen shot 2017-08-10 at 11 10 04 pm
That means if we want to get this right, we need to have the part of topocentric correction. 

-Jing Luo

**Comment by paulray** [comment]
In the docstring, you want to link to this document, page 6, which describes how this column is computed: https://ipnpr.jpl.nasa.gov/progress_report/42-196/196C.pdf

**Comment by paulray** [comment]
Your test results are exactly what I expected. So, you just have to add that topocentric term as defined on the wiki page I linked to.  That is actually the part of the transformation that I'm having trouble with for spacecraft observatories, as I note in issue #330

**Comment by luojing1211** [comment]
That means, we are lacking of the last term of Eq5 line2 from the paper. This should not be hard to do.

**Comment by luojing1211** [comment]
I added the topocentric part. The plot below is the comparison between astropy computed tdb and ephemeris computed tdb. It looks like matching our expectation. 
screen shot 2017-08-11 at 2 53 00 pm

**Comment by paulray** [comment]
Looks good! Can you say a little bit about how you did the comparison?  Were they using exactly the same ephemeris files? If the code is short maybe just post it in a comment?

**Comment by luojing1211** [comment]
I used the same ephemeris. Here is the code for doing this.
```
par = 'datafile/B1855+09_NANOGrav_9yv1.gls.par'
tim = 'datafile/B1855+09_NANOGrav_9yv1.tim'
t = toa.get_TOAs(tim, ephem='DE436t')
t2 = toa.get_TOAs(tim, ephem='DE436t', tdb_from_ephem=True)
plt.plot(t2.get_mjds(), (t.table['tdbld']-t2.table['tdbld'])*86400.0, '+')
```
This PR modified the toa.py and topo_observatory.py. Paul D. and I think this function can be more generalized and the code can be cleaner. Maybe put all of them into observatory module.

**Comment by paulray** [comment]
@demorest are you OK with merging this now, or do you think some code cleanup should be done first?

**Comment by demorest** [comment]
hi Paul, there are a couple structural things I'm not totally happy with:

1. The addition of this option as yet another "top-level" argument to get_TOAs().  The more of these we add, the kludgier it gets (in my opinion anyways).  We really should have a better system for dealing with these sorts of options.

2. The addition of an "if-then" block inside TOAs.compute_TDBs() to deal with the option.  I think things like this should probably be handled inside Observatory.

Jing and I already talked about these some when he was here last week.  I think #1 might be a bigger issue, so maybe we should just ignore that for the time being and open a separate discussion about it.  #2 should probably be addressed here but I wanted to talk with you first about how to make this handle the spacecraft case cleanly; was hoping maybe we could do this on the next telecon.

Also I have a vague suspicion we could be making better use of the astropy time framework (rather than adding cases within PINT) for handling this TT-TDB conversion.  But I haven't really had a chance to look into it.

**Comment by paulray** [comment]
@demorest yes I agree with your concerns.

Number 1 is definitely important to deal with, but is a bigger issue (as we've discussed in issue #112).

I'd be happy to discuss number 2 at a telecon. I should be able to make it on Wednesday this week. @luojing1211 can you put it on the agenda?

**Comment by luojing1211** [comment]
​Yes, I already marked it.

-Jing ​

**Comment by luojing1211** [comment]
Hi all, 
I have moved the method selection part to the observatory module. Please check it out.

**Pull request merged**